### PR TITLE
Deprecate boot2docker-cli officially

### PR DESCRIPTION
**Pull request by tianon**
This is a WIP and not ready for merge.  **DO NOT MERGE YET**.  This is being opened so we can discuss how to orchestrate and communicate the official deprecation properly, and how we'll manage the release of 1.8 to make sure users have a way to continue using this old code if they absolutely need to while still encouraging them very strongly to move off of it ASAP.

cc @ehazlett @nathanleclaire

### CODE REVIEW

**Comment by tianon** [comment]
An example of how this looks:
```console
$ ./boot2docker shellinit

  WARNING: 'boot2docker' is officially deprecated.
    Please switch to 'docker-machine' (https://github.com/docker/machine)
    as soon as possible.

error in run: Failed to get machine "boot2docker-vm": machine does not exist (Did you run `boot2docker init`?)
```

**Comment by nathanleclaire** [comment]
Yeah, that looks pretty good.  I think it might be more useful to link them to https://docs.docker.com/machine/ though.  Would at least save them a click.

**Comment by ehazlett** [comment]
Looks good -- +1 for the link to the docs as that tends to be our recommended entry point.

**Comment by tianon** [comment]
Cool, that one's an easy fix. :+1:

**Comment by tianon** [comment]
Updated the `upgrade` code to only download releases that match `1.x` or `v1.x`, and removed the "Docker client upgrade" portion as that's likely to only cause pain in the future.

File: cmds.go [hunk 1]
```diff
@@ -422,46 +411,6 @@ func upgradeBoot2DockerBinary() error {
 	return nil
 }
 
-func upgradeDockerClientBinary() error {
```
Comment (nathanleclaire) [inline cmds.go]: I'll never forget you, upgrade code 

/me _sheds single tear_

**Comment by SvenDowideit** [comment]
LGTM - can we also add a proposed EOL? (deprecation only starts a clock)

**Comment by bfirsh** [comment]
It might be worth recommending that users download Toolbox to install Machine. Nearly everyone using the boot2docker CLI will have got it from the installers, so Toolbox is the replacement from their POV.

/cc @moxiegirl @mchiang0610

**Comment by ehazlett** [comment]
SGTM

**Comment by mchiang0610** [comment]
@bfirsh +1.  Should I create a separate PR for this?

**Comment by tianon** [comment]
I'm not a big fan of the HTTP endpoint logic, but it's all in now.  I also backed out the "> v1.x" exclusion, and I've merged Nathan's PR that includes the updated language mentioning Toolbox.

**Comment by tianon** [comment]
@SvenDowideit I think an explicit EOL date will convey that we're actually still supporting this up until that point, which is a disservice to our users IMO.  We'll still "support" it, but it's going to get even less attention than it was before, so I think we're essentially saying that it's EOL _today_ barring major or severe issues.

**Comment by tianon** [comment]
image

A screenshot of a chat totally counts as a LGTM, right? :smile:

**Comment by nathanleclaire** [comment]
LOL u devious

**Comment by cblage** [comment]
is it possible to create a light boot2docker wrapper for the docker toolbox?
to facilitate migrating :D

**Comment by cblage** [comment]
like a boot2docker-cli-wrapper-docker-toolbox

**Comment by cblage** [comment]
alias boot2docker="..."

**Comment by nathanleclaire** [comment]
@cblage You might be able to set a bash function which "converts" b2d-cli commands to Docker Machine (e.g. `boot2docker up` becomes `docker-machine start default`) but it will likely be fragile and could potentially fall over in a variety of places.

**Pull request merged**